h2o-algos/src/main/java/hex/glm/ComputationState.java

```diff
@@ -215,6 +227,23 @@ public int compare(Integer o1, Integer o2) {
     return res;
   }
 
+  private static double [] extractSubRange(int N, int [] ids, double [] src, int offset) {
+    if(ids == null) return Arrays.copyOfRange(src,offset,offset+ids.length);
```

Comment:
this looks like it will cause NPE. ids == null -> ids.lenght => NPE

Reply:
Thank you Honza for the review!  Let me fix it.

Reply:
Can this really happen ? I couldn't reproduce it with current test battary. If yes, there should be a test.
Is the solution to copy the full length ? `if (ids == null) return Arrays.copyOfRange(src, offset, src.length);` ?